doc/iterations/2023.6/shapes/filterTreeBasedRepresentations.adoc

```diff
@@ -0,0 +1,92 @@
+:imagesdir: images/
+= (M) Filtering tree based representations
+
+== Problem
+
+It is quite difficult to search for/identify specific tree item(s) in a tree when this tree contains lots of tree items.
+
+== Key Result
+
+The solution should not slow-down the user interface. The user interface must remains as smooth as it was before the developments of this filter.
+The results displayed by the filter should be visible immediately, i.e. without any latency.
+Otherwise, if the solution can only be implemented with a latency, then during the computation time, a "busy" indicator should be displayed, informing the user his request is processing.
+
+== Solution
+
+Add a search capacity in the Explorer View and other tree based representations, by providing a textual filtering of the tree just like what is being done in VSCode using a simple textfield.
```

Comment:
It's not clear to me if the search/filtering only considers the tree items actually visible on the front (and thus can be fully implemented frontend-side), or if if (the search in particular) will also potentially reveal new elements (by asking the backend).

I think the proposal is for the first option, but unless I missed something this is never mentioned explicitly.

"like what is being done in VSCode" in particular can be confusing. In VS Code, even if all my projects are collapsed in the Explorer, it will look for my search term as deep as needed and expand new folders to reveal the matching files.

A purely frontend-based "search" feature look like it could confuse users. If I search for "foo" in my project, why would it "find" it only when the corresponding item was already expanded/visible, and then a few minutes later, just because I have collapsed the part of the tree that contains it the same "search" does not find it.
As a user, it means that if I search for something in my project and Sirius Web does not give me any match, I can not assume the element does not exist. Maybe it's present somewhere deep inside a hierarchy I have not expanded yet? But is it not the whole purpose of a search to find that for me?

Reply:
Actually it is mentioned in the very last sentence of the "No go" section that all of this only considers visible tree items.
I think the term "search" should not be used as it gives wrong expectations.